### ISSUE TITLE
Compile SDK / Target SDK version not consistent with defaults in root build.gradle

### ISSUE THREAD

**Issue by brodybits** [apache/cordova-android#631]
If I would do `cordova platform add android@nightly` in a new Cordova project and then do `cordova build android`, the APK has target SDK version 27 while SDK 28 is specified as default in: <https://github.com/apache/cordova-android/blob/f1f1ac3cbd7e75aa1dfeb6faf856a90335d2197c/bin/templates/project/build.gradle#L45-L46>

I used the following command to check the SDK versions in the APK:

```sh
aapt list -a platforms/android/app/build/outputs/apk/debug/app-debug.apk | grep SdkVersion
```

This issue *may* be related to #508 - Inconsistent handling of min/target SDK values

I hope we can resolve this one before the next major release for Cordova 9 (apache/cordova#10).

P.S. I expanded the scope of this issue to cover the compile SDK version. This issue needs to be resolved before we can continue with AndroidX support (#565).

**Comment by brodybits** [comment]
I think this should be done **before** we try to resolve the issue with AndroidX (#565).

I think this issue is similar to #508 (*Inconsistent handling of min/target SDK values*) and needs to be solved in the Gradle scripts. The solution *may* be related to PR #633 (quick fix for #508).

/cc @davidschreiber

**Comment by brodybits** [comment]
I am expanding the scope of this issue to cover the compile SDK version, as @davidschreiber pointed out in <https://github.com/apache/cordova-plugin-camera/pull/418#issuecomment-459266227> and <https://github.com/apache/cordova-plugin-camera/pull/418#issuecomment-459272388>. As I said before:

- I think this is similar to and maybe overlaps #508 (*Inconsistent handling of min/target SDK values*).
- A solution similar to PR #633 (quick fix for #508) may be needed here.
- Needs to be resolved before we continue with AndroidX support (#565)

**Comment by erisu** [comment]
@brodybits I think we can close this ticket now as PR #699 is merged into master.
Is my assumption correct?

**Comment by brodybits** [comment]
I am not yet convinced that PR #699 resolves this issue, which I think is better described by #658.

I would like to test it to be absolutely sure.

Unfortunately I cannot promise when I will get a chance to test it due to some other priorities. Leaving it up to your best judgment for now.

**Comment by erisu** [comment]
@brodybits I can continue to look into this. 

I have a plan for another PR which will refactor the Gradle process but the amount of work might not fit into this release version. I will see what can be done in patch/minor releases and keep the major changes in a seperate  PR.

From the ticket, you linked

> the following default values seem to be ignored:
> 
> * defaultBuildToolsVersion
> * defaultTargetSdkVersion
> * defaultCompileSdkVersion

`defaultTargetSdkVersion` and `defaultCompileSdkVersion` should be used now from my last PR.

`defaultBuildToolsVersion` was the only one I left out as I was looking at it around the Gradle refactor planning.

**Comment by erisu** [comment]
For this ticket, I can do some additional testing around Target and Compiled SDK.
As for the build tool version, that will be handled in the other ticket.

**Comment by erisu** [comment]
Closing this ticket as resolved in Cordova-Android 10.

https://github.com/apache/cordova-android/pull/1212
